lib/ansible/modules/get_url.py

```diff
@@ -444,7 +444,7 @@ def extract_filename_from_headers(headers):
 
     Looks for the content-disposition header and applies a regex.
     Returns the filename if successful, else None."""
-    cont_disp_regex = 'attachment; ?filename="?([^"]+)'
+    cont_disp_regex = 'attachment|inline; ?filename="?([^"]+)'
```

Comment:
While this may fix a specific issue this could still have an undesired behavior when the [type is unknown](https://datatracker.ietf.org/doc/html/rfc6266#section-4.2).

Since we don't really care about the type we don't have to query for it and just retrieve the filename value. Unfortunately it seems that we cannot use [`ansible.module_utils.urls.get_response_filename`](https://github.com/ansible/ansible/blob/ff5deaf62fa7c4fd610fb77c80d02ca8311e8f33/lib/ansible/module_utils/urls.py#L342) as we don't have the response object, we can probably just utilize [`email.message.Message.get_filename()`](https://docs.python.org/3/library/email.message.html#email.message.EmailMessage.get_filename) to parse the filename instead of manually doing the work ourselves.